core/types/types.cc

```diff
@@ -1024,7 +1031,7 @@ TypePtr Types::applyTypeArguments(const GlobalState &gs, const CallLocs &locs, u
     }
 
     if (genericClass.data(gs)->typeMembers().empty()) {
-        return Types::untypedUntracked();
+        return Types::untyped(Symbols::Magic_UntypedSource_applyTypeArguments());
```

Comment:
I believe this is actually a case where we _would_ want to blame to an error recovery symbol.

I believe this case can only happen if you attempt to provide type arguments to a generic class that doesn't accept type arguments. That error is reported above, and this condition recovers from it.